src/tests/ie_test_utils/common_test_utils/graph_comparator.cpp

```diff
@@ -313,7 +313,30 @@ class BackEdge {
               m_result(not_null(result)) {}
 
     bool result_and_parameter_match() const {
-        return equal_type_and_partial_shape(m_result->output(0), *m_parameter);
+        if (m_parameter->get_element_type() != m_result->output(0).get_element_type()) {
+            return false;
+        }
+        const auto& param_shape = m_parameter->get_partial_shape();
+        const auto& result_shape = m_result->output(0).get_partial_shape();
+        const auto& param_rank = param_shape.rank();
+        const auto& result_rank = result_shape.rank();
+        if (!param_rank.compatible(result_rank)) {
+            return false;
+        }
+        if (param_rank.is_static() && result_rank.is_static()) {
```

Comment:
Currently `result_and_parameter_match()` will return `true` also for pair of dynamic and static rank. Is it expected?

Reply:
I think so, especially param can have static rank, but we cannot calculate result rank. `param_rank.compatible(result_rank)` is adequate for me here